mri/reconstruct/utils.py

```diff
@@ -233,3 +233,46 @@ def generate_operators(data, wavelet_name, samples, mu=1e-06, nb_scales=4,
         prox_op=prox_op,
         verbose=verbose)
     return gradient_op, linear_op, prox_op, cost_op
+
+
+def get_stacks_fourier(samples):
+    """ Function that converts an incoming 3D kspace samples
+        and converts to stacks of 2D. This function also checks for
+        any issues of the incoming k-space pattern and if the stack property
+        is not satisfied.
+        Stack Property:
+            The k-space locations originate from a stack of 2D samples
+    Parameters
+    ----------
+    samples: np.ndarray
+        the mask samples in the 3D Fourier domain.
+
+    Returns
+    ----------
+    plane_samples: np.ndarray
+        A 2D array of samples which when stacked gives the 3D samples
+    z_samples: np.ndarray
+        A 1D array of z-sample locations
```

Comment:
can you call it `z_samples_loc` then?

Reply:
Yes. Ill use `z_sample_loc` and `kspace_plane_loc`